gremlin-server/src/test/java/org/apache/tinkerpop/gremlin/util/Log4jRecordingAppender.java

```diff
@@ -63,9 +64,18 @@ public void clear() {
         messages.clear();
     }
 
+    /**
+     * @param regex not null
+     * @return true if there is a substring of a message matching the regular expression, where:
+     *         . matches also the EOLN char(s) defined in the layout.
+     *         $ matches the end of the string
+     */
     public boolean logContainsAny(final String regex) {
-        // chop off the line feed so that the regex doesn't have to account for that
-        return messages.stream().anyMatch(m -> m.substring(0,m.length() - 1).matches(regex));
+        Pattern pattern;
```

Comment:
minor nit - could you inline initialize `pattern` and mark it `final` - note that our typical style is to `final` all variables that don't mutate. if you could give a general review to your changes to fix those where necessary that would be great.